xla/pjrt/pjrt_stream_executor_client.cc

```diff
@@ -3347,6 +3347,10 @@ PjRtStreamExecutorClient::Compile(const XlaComputation& computation,
                                   CompileOptions options) {
   tsl::profiler::TraceMe traceme("PjRtStreamExecutorClient::Compile");
   VLOG(1) << "PjRtStreamExecutorClient::Compile";
+  options.executable_build_options.set_process_index(process_index());
+  // Assuming that each process has the same number of devices.
+  options.executable_build_options.set_process_count(
```

Comment:
maybe TF_RET_CHECK that it's divisible? Or alternatively remove set_process_count from this PR, since this line makes some assumptions/choices, while the rest of the PR is plumbing information through.